File: Manuel/Class 3/ATM/atm.py
import sys

def solve():
    N = int(sys.stdin.readline())
    times = list(map(int, sys.stdin.readline().split()))
    times.sort()

    total_waiting_time = 0
    current_time = 0

    for time_needed in times:
        current_time += time_needed
        total_waiting_time += current_time
    
    print(total_waiting_time)

solve()

File: Manuel/개인 활동/Class 3/ATM/atm.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin >> N;
    vector<int> times(N);
    for(int i = 0; i < N; i++)
    {
        cin >> times[i];
    }
    sort(times.begin(), times.end());

    long long total_waiting_time = 0;
    long long current_time = 0;

    for(int i = 0; i < N; i++)
    {
        current_time += times[i];
        total_waiting_time += current_time;
    }
    
    cout << total_waiting_time << "\n";

    return 0;
}